streamingpro-mlsql/src/main/java/tech/mlsql/datasource/impl/MLSQLRest.scala

```diff
@@ -73,153 +74,101 @@ class MLSQLRest(override val uid: String) extends MLSQLSource
   override def load(reader: DataFrameReader, config: DataSourceConfig): DataFrame = {
     val skipParams = config.config.getOrElse("config.page.skip-params", "false").toBoolean
     val retryInterval = JavaUtils.timeStringAsMs(config.config.getOrElse("config.retry.interval", "1s"))
-
     val debug = config.config.getOrElse("config.debug", "false").toBoolean
+    val maxTries = config.config.getOrElse("config.retry", config.config.getOrElse("config.page.retry", "3")).toInt
 
-    (config.config.get("config.page.next"), config.config.get("config.page.values")) match {
-      case (Some(urlTemplate), Some(jsonPath)) =>
-        val maxSize = config.config.getOrElse("config.page.limit", "1").toInt
-        val maxTries = config.config.getOrElse("config.page.retry", "3").toInt
-        val pageInterval = JavaUtils.timeStringAsMs(config.config.getOrElse("config.page.interval", "10ms"))
-        var count = 1
+    def _error2DataFrame(errMsg: String, statusCode: Int, session: SparkSession) : DataFrame = {
+      session.createDataFrame(session.sparkContext.makeRDD(Seq(Row.fromSeq(Seq( errMsg.getBytes(UTF_8), 500))))
+        , StructType(fields = Seq(
+          StructField("content", BinaryType), StructField("status", IntegerType)
+        )))
+    }
 
-        val pageStrategy = PageStrategyDispatcher.get(config.config)
-        if (debug) {
-          logInfo(format(s"Started to get Page ${count} ${config.path} "))
+    /**
+     * Calling http rest endpoints with retrying
+     * @param url
+     * @return http status code along with DataFrame
+     */
+    def _httpWithRetrying(url: String, skipParams: Boolean): (Int, DataFrame) = {
+      executeWithRetrying[(Int, DataFrame)](maxTries)((() => {
```

Comment:
How can we distinct `config.page.retry` and ``config.retry`, since we all use maxTries.

Reply:
Code is changed, use `config.pae.retry` for multi-page request; `config.page.retry` and `config.retry` for single-page request.